jbmc/src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -3184,28 +3184,38 @@ void java_bytecode_convert_methodt::save_stack_entries(
 {
   for(auto &stack_entry : stack)
   {
+    // remove typecasts if existing
+    auto entry = std::ref(stack_entry);
+    while(entry.get().id() == ID_typecast)
+    {
+      entry = std::ref(to_typecast_expr(stack_entry).op());
+    }
```

Comment:
There already is code doing `skip_typecast`, but that's just local. I have created #2508 to change that.